subsys/net/lib/http/http_app_server.c

```diff
@@ -180,6 +181,14 @@ int http_send_error(struct http_ctx *ctx, int code,
 	case 400:
 		msg = HTTP_STATUS_400_BR;
 		break;
+	default:
+		if (code < 100 || code > 999) {
+			return -EINVAL;
+		}
+
+		snprintk(buf, sizeof(buf), "HTTP/1.1 %3d\r\n\r\n", code);
```

Comment:
Don't you also need to send a text after the numerical code? `200 OK` or `404 Not Found`? Most clients will ignore this, but this isn't compliant.  See [RFC2616, section 6.1](https://tools.ietf.org/html/rfc2616#section-6.1).

Reply:
There is no string to print atm as only 400 code is having a error string defined.

Reply:
If you think we need the error string too, then we need to fix the API (which is ok in this case as it was merged just one week ago).